### PR TITLE
Vary ATI domain depending on environment

### LINKED ISSUE #2001: Update ATI domain on Test
**Is your feature request related to a problem? Please describe.**
As a result of testing, we've found that the ATI base url on Test needs to be updated from `https://a1.api.bbc.co.uk/hit.xiti` to `https://logws1363.ati-host.net`

**`localhost` should use the test URL!**

It should not change for the live environment. 

**Describe the solution you'd like**
- Update the ATIAnalytics container to change the base url depending on the environment param `env`. 
- Update unit and cypress tests.

## DO NOT MERGE
**Prior to this being merged, will need to ensure that `https://logws1363.ati-host.net` is added to CSP headers on our Test environment** https://github.com/bbc/simorgh-infrastructure/issues/512

**Describe alternatives you've considered**
N/A

**Testing notes**
- Build branch to test to ensure CSP is in place.
- Observe new domain for ATI

Dev insight: Unit and cypress tests

### PR DESCRIPTION
**Pull request by sareh**
Resolves #2001

**Overall change:** When on local or test environments, the ATI url is updated 
to `https://logws1363.ati-host.net` on live it should remain as `https://a1.api.bbc.co.uk/hit.xiti`

**Code changes:**

- Update cypress tests
- Ensure RequestContext is passing in the env as a prop
- Use the env from the RequestContext in `ATIAnalytics/canonical` & `ATIAnalytics/amp` variants
- Update atiBaseUrl to be a function that varies on env. 

Using the ATI Chrome browser plugin, I see all of the correct attributes, as well as the request made to the `https://logws1363.ati-host.net` domain for both [canonical](http://localhost.bbc.co.uk:7080/news/articles/c0g992jmmkko) and [amp pages](http://localhost.bbc.co.uk:7080/news/articles/c0g992jmmkko.amp)

Screen Shot ati domain

For the live environment, run the instructions here in the readme to get the live bundle running locally: 
Then you should see the request being made to the live endpoint in the console
Screen Shot 2019-06-26 at 17 50 11

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Comment by dr3** [comment]
This will break stage ATI, as stage is using live mozart CSP rules, but simorgh uses test ATI for stage, it will break :(

Can we make stage use Live ATI?

**Comment by sareh** [comment]
After this is merged & deployed to Test, you should no longer see this CSP header error on the Test environment:
Screen Shot 2019-07-05 at 17 47 17
Since we should be making the request to `https://logws1363.ati-host.net` on Test.

**Comment by sareh** [comment]
@dr3 Stage will be using the live endpoint, since we only explicitly use the Test ATI endpoint if `env` is `local` or `test`.

**Comment by dr3** [comment]
> @dr3 Stage will be using the live endpoint, since we only explicitly use the Test ATI endpoint if `env` is `local` or `test`.

@sareh Will it be using the live destination though? i think it uses the test one. Can you use the test destination on the live url?

File: src/app/containers/ATIAnalytics/atiUrl.js [hunk 1]
```diff
@@ -12,7 +12,12 @@ import {
   sanitise,
 } from '../../lib/analyticsUtils';
 
-export const atiBaseUrl = 'https://a1.api.bbc.co.uk/hit.xiti?';
+export const atiBaseUrl = env => {
+  if (env === 'local' || env === 'test') {
+    return 'https://logws1363.ati-host.net?';
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/atiUrl.js]: Howcome these urls are in code, and in the .env files?

**Review by dr3 — CHANGES_REQUESTED** [review]
Can you change https://github.com/bbc/simorgh/blob/latest/src/app/contexts/RequestContext/getStatsDestination/index.js#L10 to be 'live' || 'stage' please? 

This PR makes stage use the LIVE url, but it still uses the TEST destinations

**Comment by 12** [comment]
Blocked pending #2409

File: src/app/containers/ATIAnalytics/amp/index.test.jsx [hunk 2]
```diff
@@ -11,16 +12,54 @@ describe('Amp ATI Analytics', () => {
     document.body.appendChild(container);
   });
 
-  it('should render amp-analytics with script tag', () => {
-    act(() => {
-      ReactDOM.render(<AmpATIAnalytics />, container);
-    });
+  const testCases = [
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/amp/index.test.jsx]: In these tests personally ide prefer if you mocked `useContext` to return `env` that you set. Then mock `atiBaseUrl(env)`, expect it to have been called with `env` and then mock to return a fake value. This test tests the functionality of `getEnv()` and `atiBaseUrl()` alot

File: src/app/containers/ATIAnalytics/canonical/index.jsx [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
 
   useEffect(() => {
     sendBeacon(atiPageViewUrl);
   }, [atiPageViewUrl]);
 
   return (
-    <noscript>
-      <img height="1px" width="1px" alt="" src={atiPageViewUrl} />
-    </noscript>
+    <noscript
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/index.jsx]: Howcome the switch to dangerouslyinnerset?

File: src/app/containers/ATIAnalytics/canonical/index.test.jsx [hunk 4]
```diff
@@ -23,36 +23,110 @@ const ContextWrap = ({ children }) => (
...
+  });
   afterEach(() => {
+    document.body.removeChild(container);
+    container = null;
     jest.clearAllMocks();
   });
+  const mockPageviewParams = 'key=value&key2=value2';
+
+  // following tests are commented out so can debug them one at a time
+  const testCases = [
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/index.test.jsx]: These tests are also verrrry integationy, but might let it slide

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/ATIAnalytics/canonical/index.jsx (hunk 3 above)
Comment (12) [reply src/app/containers/ATIAnalytics/canonical/index.jsx]: `noscript` was being rendered with no `innerHTML` being set previously.

Comment (12) [reply src/app/containers/ATIAnalytics/canonical/index.jsx]: Have fixed this.

File: src/app/containers/ATIAnalytics/amp/index.test.jsx (hunk 2 above)
Comment (12) [reply src/app/containers/ATIAnalytics/amp/index.test.jsx]: Good idea, have done this

**Review by dr3 — APPROVED** [review]

**Comment by jamesdonoh** [comment]
Once this work is released @rhell please can you verify that we no longer have console errors and if so let the infra-tooling pod know they can start work on https://github.com/bbc/simorgh-infrastructure/issues/522

**Comment by 12** [comment]
> Once this work is released @rhell please can you verify that we no longer have console errors and if so let the infra-tooling pod know they can start work on [bbc/simorgh-infrastructure#522](https://github.com/bbc/simorgh-infrastructure/issues/522)

Good shout @jamesdonoh , yep will definitely make sure that happens 🙂

File: src/app/containers/ATIAnalytics/atiUrl.js [hunk 5]
```diff
@@ -12,7 +12,12 @@ import {
...
+  }
+  return 'https://a1.api.bbc.co.uk/hit.xiti?';
```
Comment (twinlensreflex) [inline src/app/containers/ATIAnalytics/atiUrl.js]: does it make more sense to store these values in envConfig? wouldn't need if/else logic then either, right?
e.g
```suggestion
  return process.env.SIMORGH_ATI_BASEURL;
```

Comment (pjlee11) [reply src/app/containers/ATIAnalytics/atiUrl.js]: Based on this do we need the if statement? Surely we can have the local and test config have `SIMORGH_ATI_BASEURL` switch between `https://logws1363.ati-host.net?` and `https://a1.api.bbc.co.uk/hit.xiti?` based on the env config?

File: src/app/containers/ATIAnalytics/atiUrl.js [hunk 6]
```diff
@@ -12,7 +12,7 @@ import {
   sanitise,
 } from '../../lib/analyticsUtils';
 
-export const atiBaseUrl = 'https://a1.api.bbc.co.uk/hit.xiti?';
+export const atiBaseUrl = () => process.env.SIMORGH_ATI_BASE_URL;
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/atiUrl.js]: Does this need to be a function? Cant you just call `process.env.SIMORGH_ATI_BASE_URL` where its used? Also then it doesnt need isolated tests. You can stub it in tests by just setting `process.env.SIMORGH_ATI_BASE_URL` in the tests

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/ATIAnalytics/FrontPageAtiParams.jsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import { useContext } from 'react';
-import { atiPageViewParams } from './atiUrl';
+import atiPageViewParams from './atiUrl';
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/FrontPageAtiParams.jsx:2]: Doesn't feel like `atiUrl` is a good name anymore 🙈

File: src/app/containers/ATIAnalytics/canonical/index.jsx [hunk 8]
```diff
@@ -1,10 +1,11 @@
 import React, { useEffect, useState } from 'react';
 import { string } from 'prop-types';
-import { atiBaseUrl } from '../atiUrl';
 import sendBeacon from '../../../lib/analyticsUtils/sendBeacon';
 
 const CanonicalATIAnalytics = ({ pageviewParams }) => {
-  const [atiPageViewUrl] = useState(atiBaseUrl + pageviewParams);
+  const [atiPageViewUrl] = useState(
+    process.env.SIMORGH_ATI_BASE_URL + pageviewParams,
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/index.jsx:7]: some might want this to be `${process.env.SIMORGH_ATI_BASE_URL}${pageviewParams}`, for the record, im not one of them :P

**Review by dr3 — APPROVED** [review]
LGTM

**Review by rhenshaw56 — APPROVED** [review]
Looks good!

**Comment by jamesbrumpton** [comment]
LGTM

**Pull request merged**